crates/storage_models/src/query/generics.rs

```diff
@@ -111,6 +111,66 @@ where
         .attach_printable_lazy(|| format!("Error while updating {debug_values}"))
 }
 
+#[instrument(level = "DEBUG", skip_all)]
+pub async fn generic_update_with_unique_predicate_get_result<T, V, P, R>(
```

Comment:
merchant_id is a unique index, but if it was PK. Then this generic function can be avoided.

Reply:
making unique identifier of a table to primary key we will take it up in a separate PR. After this we will deprecate this